pkg/agent/memberlist/cluster.go

```diff
@@ -252,7 +252,7 @@ func (c *Cluster) enqueueExternalIPPool(obj interface{}) {
 // newClusterMember gets the Node's IP and returns a cluster member "<IP>:<clusterMemberlistPort>"
 // representing that Node in the memberlist cluster.
 func (c *Cluster) newClusterMember(node *corev1.Node) (string, error) {
-	nodeAddr, err := k8s.GetNodeAddr(node)
+	nodeAddr, err := k8s.GetNodeAddrs(node)
```

Comment:
Used `nodeAddrs.IPv4` as IP of `member`.

There's no test for this?

Reply:
There are tests for it and they are expected to fail because the Nodes won't be able to connect to other Nodes and all of them will assign the egress IPs to themselves. I guess the failure is not detected because the verification didn't check duplicate assignment.